python/semantic_kernel/functions/kernel_function_decorator.py

```diff
@@ -53,61 +54,66 @@ def decorator(func: Callable):
         func.__kernel_function_parameters__ = [
             _parse_parameter(param) for param in func_sig.parameters.values() if param.name != "self"
         ]
-
+        return_param_dict = {}
         if func_sig.return_annotation != Signature.empty:
-            return_description, return_type, return_required = _parse_annotation(func_sig.return_annotation)
-        else:
-            return_description, return_type, return_required = "", "None", False
-        func.__kernel_function_return_type__ = return_type
-        func.__kernel_function_return_description__ = return_description
-        func.__kernel_function_return_required__ = return_required
+            return_param_dict = _parse_annotation(func_sig.return_annotation)
+        func.__kernel_function_return_type__ = return_param_dict.get("type_", "None")
+        func.__kernel_function_return_description__ = return_param_dict.get("description", "")
+        func.__kernel_function_return_required__ = return_param_dict.get("is_required", False)
         return func
 
     return decorator
 
 
-def _parse_parameter(param: Parameter):
+def _parse_parameter(param: Parameter) -> Dict[str, Any]:
     logger.debug(f"Parsing param: {param}")
-    param_description = ""
-    type_ = "str"
-    required = True
+    ret = {}
     if param != Parameter.empty:
-        param_description, type_, required = _parse_annotation(param.annotation)
-    logger.debug(f"{param_description=}, {type_=}, {required=}")
-    return {
-        "name": param.name,
-        "description": param_description,
-        "default_value": param.default if param.default != Parameter.empty else None,
-        "type": type_,
-        "required": required,
-    }
+        ret = _parse_annotation(param.annotation)
+    ret["name"] = param.name
+    if param.default != Parameter.empty:
+        ret["default_value"] = param.default
+    return ret
 
 
-def _parse_annotation(annotation: Parameter) -> Tuple[str, str, bool]:
+def _parse_annotation(annotation: Parameter) -> Dict[str, Any]:
```

Comment:
Would it make sense to create a class for this return value? Given that it has a defined structure maybe it could benefit from being a defined and typed object.

Reply:
Or is that what `KernelParameterMetadata` is for? Confusing because of the `Dict` being used across this function.

Reply:
yeah this will be stored in the __kernel_function_parameters__ field of a function, so I don't want to make that more heavy then necessary, when creating a actual KernelFunctionFromMethod these get parsed into KernelParameterMetadata.